### ISSUE TITLE
Extreme high prediction time

### ISSUE THREAD

**Issue by LenaMetzen** [openvinotoolkit/anomalib#1452]
### Describe the bug

I first trained an efficient_ad model with my own dataset. Afterwards I want to make some prediction. But i got the problem, that the time needed to make one prediction in extremely high. One prediction takes between 90s and 120s. How can I improve the time?

GPU: NVIDIA GeForce RTX 2060

### Dataset

Other (please specify in the text field below)

### Model

Other (please specify in the field below)

### Steps to reproduce the behavior

1. Train model
2. Run inference
```
for file in test_images_paths:
    image_name = str(file).split("/")[-1].split(".")[0]
    image_name = image_name + ".png"

    command = 'python tools/inference/lightning_inference.py --config results/efficient_ad_SolderJoints_reduzierte_Anzahl_100_Epochs/efficient_ad/SolderJoint/run/config.yaml --weights results/efficient_ad_SolderJoints_reduzierte_Anzahl_100_Epochs/efficient_ad/SolderJoint/run/weights/lightning/model-v1.ckpt --input datasets/SolderJoints/test_eindeutig_sortiert/' + image_name + ' '
    command = command + '--output run/efficient_ad_solder_joints --visualization_mode "simple"'# --show'
    command = command.replace("\\", "/")
    timeStart = time.time()
    subprocess.run(str(command), shell=True)
    timePred = "Time for prediction: {:5.3f}s".format(time.time() - timeStart)
    print(timePred)
```

### OS information

OS information:
- OS: [e.g. Ubuntu 20.04]
- Python version: [e.g. 3.8.10]
- Anomalib version: [e.g. 0.3.6]
- PyTorch version: [e.g. 1.9.0]
- CUDA/cuDNN version: [e.g. 11.1]
- GPU models and configuration: [e.g. 2x GeForce RTX 3090]
- Any other relevant information: [e.g. I'm using a custom dataset]

### Expected behavior

Faster inference speed

### Screenshots

![Capture](https://github.com/openvinotoolkit/anomalib/assets/101172221/24bcbec2-c8c8-4c52-bc2b-52b5223cc854)

### Pip/GitHub

pip

### What version/branch did you use?

anomalib                  1.0.0.dev0

### Configuration YAML

```yaml
dataset:
  name: SolderJoint
  format: folder
  root: ./datasets/images/
  normal_dir: IO
  abnormal_dir: NIO
  task: classification
  mask_dir: null
  normal_test_dir: null
  extensions: null
  split_ratio: 0.2
  seed: 0
  image_size: 256
  train_batch_size: 64
  eval_batch_size: 64
  num_workers: 24
  normalization: none
  test_split_mode: from_dir
  val_split_ratio: 0.5
  transform_config:
    eval: null
    train: null
  val_split_mode: from_test
logging:
  log_graph: False
  logger: [] 
metrics:
  image:
  - F1Score
  - AUROC
  threshold:
    manual_image: null
    manual_pixel: null
    method: adaptive
model:
  lr: 0.0001
  model_size: small
  name: efficient_ad
  normalization_method: min_max
  pad_maps: True
  padding: False
  teacher_out_channels: 38
  weight_decay: 0.00001
optimization:
  export_mode: null
project:
  path: ./results/efficient_ad
  seed: 42
trainer:
  accelerator: gpu
  accumulate_grad_batches: 1
  auto_lr_find: False
  auto_scale_batch_size: False
  benchmark: False
  check_val_every_n_epoch: 1
  default_root_dir: null
  detect_anomaly: False
  deterministic: False
  devices: 1
  enable_checkpointing: True
  enable_model_summary: True
  enable_progress_bar: True
  fast_dev_run: False
  gradient_clip_algorithm: norm
  gradient_clip_val: 0
  limit_predict_batches: 1.0
  limit_test_batches: 1.0
  limit_train_batches: 1.0
  limit_val_batches: 1.0
  log_every_n_steps: 50
  max_epochs: 50
  max_steps: 70000
  max_time: null
  min_epochs: null
  min_steps: null
  move_metrics_to_cpu: False
  multiple_trainloader_mode: max_size_cycle
  num_nodes: 1
  num_sanity_val_steps: 0
  overfit_batches: 0.0
  plugins: null
  precision: 32
  profiler: null
  reload_dataloaders_every_n_epochs: 0
  replace_sampler_ddp: True
  strategy: null
  sync_batchnorm: False
  track_grad_norm: -1
  val_check_interval: 1.0
visualization:
  image_save_path: null
  log_images: False
  mode: full
  save_images: true
  show_images: False # show images on the screen
```
```


### Logs

```shell
NVIDIA GeForce RTX 2060
C:\Users\z004uhnp\.conda\envs\anomalib_env\lib\site-packages\anomalib\config\config.py:280: UserWarning: config.project.unique_dir is set to False. This does not ensure that your results will be written in an empty directory and you may overwrite files.
  warn(
C:\Users\z004uhnp\.conda\envs\anomalib_env\lib\site-packages\torchmetrics\utilities\prints.py:36: UserWarning: Metric `PrecisionRecallCurve` will save all targets and predictions in buffer. For large 
datasets this may lead to large memory footprint.
  warnings.warn(*args, **kwargs)
C:\Users\z004uhnp\.conda\envs\anomalib_env\lib\site-packages\anomalib\utils\callbacks\__init__.py:142: UserWarning: Export option: None not found. Defaulting to no model export
  warnings.warn(f"Export option: {config.optimization.export_mode} not found. Defaulting to no model export")
C:\Users\z004uhnp\.conda\envs\anomalib_env\lib\site-packages\pytorch_lightning\trainer\connectors\checkpoint_connector.py:55: LightningDeprecationWarning: Setting `Trainer(resume_from_checkpoint=)` is deprecated in v1.5 and will be removed in v2.0. Please pass `Trainer.fit(ckpt_path=)` directly instead.
  rank_zero_deprecation(
GPU available: True (cuda), used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
`Trainer(limit_train_batches=1.0)` was configured so 100% of the batches per epoch will be used..
`Trainer(limit_val_batches=1.0)` was configured so 100% of the batches will be used..
`Trainer(limit_test_batches=1.0)` was configured so 100% of the batches will be used..
`Trainer(limit_predict_batches=1.0)` was configured so 100% of the batches will be used..
`Trainer(val_check_interval=1.0)` was configured so validation will run at the end of the training epoch..
C:\Users\z004uhnp\.conda\envs\anomalib_env\lib\site-packages\torchmetrics\utilities\prints.py:36: UserWarning: Metric `ROC` will save all targets and predictions in buffer. For large datasets this may lead to large memory footprint.
  warnings.warn(*args, **kwargs)
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]
C:\Users\z004uhnp\.conda\envs\anomalib_env\lib\site-packages\pytorch_lightning\trainer\connectors\data_connector.py:224: PossibleUserWarning: The dataloader, predict_dataloader 0, does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` (try 24 which is the number of cpus on this machine) in the `DataLoader` init to improve performance.
  rank_zero_warn(
Predicting DataLoader 0: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:31<00:00, 31.36s/it]
Time for prediction: 60.973s
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by blaz-r** [comment]
Hello.
There might be more reasons for this, but the first thing is that this is not really just inference time. The time you track also includes time needed to create instances of dataloader, trainer, metrics and many other things. In real scenario, you usually don't re-instantiate all those things for prediction, a more realistic approach would be to just time `predict` step.
https://github.com/openvinotoolkit/anomalib/blob/66532fc1625020ce7ff6f3fc1565663aa540e834/tools/inference/lightning_inference.py#L83
Now this still doesn't explain such a long duration. I believe the issue is having 24 workers for the dataloader, so try using `num_workers: 0`. 
Having more workers is usually better when you are sending multiple data to the model. However this is not the case for small datasets or a single image. I also noticed that on some of my systems, having this number high results in a very long delay until the first step of training. I'm not sure why, but it might be due to the way the dataloader copies are made and orchestrated behind the scenes.  

To summarize: try setting `num_workers` to 0 and keep in mind that the way you benchmark doesn't capture just the inference time.

**Comment by samet-akcay** [comment]
Hi @LenaMetzen, you could refer to #1183. I'm closing this as this is a duplicate. Feel free to ask questions there if you experience any issues. Good luck!
